### PR TITLE
Sqlite

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
This PR adds functionality for querying SQLite3, see discussion on [zulip](https://roc.zulipchat.com/#narrow/stream/304641-ideas/topic/basic-webserver.20Sqlite3/near/420178165). 

The current API includes an `execute` task which is minimal but covers most use cases.

```ruby
# Definition in "platform/SQLite3.roc"
execute :
    {
        path : Str,
        query : Str,
        bindings : List Binding,
    }
    -> Task (List (List Value)) Error

# Query todos table
rows <-
    SQLite3.execute {
        path: maybeDbPath |> Result.withDefault "<DB_PATH> not set on environment",
        query: "SELECT id, task FROM todos WHERE status = :status;",
        bindings: [{ name: ":status", value: "completed" }],
    }
    |> Task.map \rows ->
        # Parse each row into a string
        List.map rows \cols ->
            when cols is
                [Integer id, String task] -> "row $(Num.toStr id), task: $(task)"
                _ -> crash "unexpected values returned, expected Integer and String got $(Inspect.toStr cols)"
    |> Task.onErr \err ->
        # Crash on any errors for now
        crash "$(SQLite3.errToStr err)"
    |> Task.await
```

The SQLIite3 values supported are;

```ruby
SQLiteValue : [
    Null,
    Real F64,
    Integer I64,
    String Str,
    Bytes (List U8),
]
```

Connections are cached in the platform so that they can be reused between calls.

Here is an example of parsing the rows returned by a query

```ruby
parseCompletedTodos : List (List SQLite3.Value) -> List Str
parseCompletedTodos = \rows ->
    rows
    |> List.map \cols ->
        when cols is
            [Integer id, String task] -> "row $(Num.toStr id), task: $(task)"
            _ -> crash "unexpected values returned, expected Integer and String got $(Inspect.toStr cols)"
```

### CODE REVIEW

**Comment by lukewilliamboswell** [comment]
It looks much larger than it actually is because I re-generated glue for the sqlite types, and also `roc format` on the platform which includes a lot more files than are actually changed.

File: examples/sqlite3.roc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    # Read DB_PATH environment variable
+    maybeDbPath <- Env.var "DB_PATH" |> Task.attempt
+
+    # Query todos table
+    rows <-
+        SQLite3.execute {
+            path: maybeDbPath |> Result.withDefault "<DB_PATH> not set on environment",
+            query: "SELECT id, task FROM todos WHERE status = :status;",
+            bindings: [{ name: ":status", value: "completed" }],
```
Comment (rtfeldman) [inline examples/sqlite3.roc:23]: I dunno if this has been discussed elsewhere, but it should be possible to have the bindings look like `{ status: "completed" }` and then have an `Encoding` implementation which takes care of translating them to the equivalent of this! 😃

File: examples/sqlite3.roc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        }
+        |> Task.map \rows ->
+            # Parse each row into a string
+            List.map rows \cols ->
+                when cols is
+                    [Integer id, String task] -> "row $(Num.toStr id), task: $(task)"
+                    _ -> crash "unexpected values returned, expected Integer and String got $(Inspect.toStr cols)"
```
Comment (rtfeldman) [inline examples/sqlite3.roc:30]: I also suspect we can make decoding for this Just Work too!

**Review by rtfeldman — DISMISSED** [review]
Looks great, thanks so much @lukewilliamboswell! 😃 

There are some merge conflicts, but otherwise this seems good to merge!

**Comment by Anton-4** [comment]
I took care of the merge conflicts :)

**Comment by lukewilliamboswell** [comment]
> I took care of the merge conflicts :)

Thank you Anton, I hadn't had a chance to update. Love that this is in.

**Pull request merged**